internal/core/src/storage/ThreadPools.h

```diff
@@ -0,0 +1,60 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MILVUS_THREADPOOLS_H
+#define MILVUS_THREADPOOLS_H
+
+#include "ThreadPool.h"
+#include "common/Common.h"
+
+namespace milvus {
+
+enum ThreadPoolPriority {
+    HIGH = 0,
+    MIDDLE = 1,
+    LOW = 2,
+};
+
+class ThreadPools {
+public:
+    static ThreadPools& GetInstance() {
```

Comment:
FYI, this could be simplified, to pass priority as parameter of GetInstance, and maintains a static `std::map<ThreadPoolPriority, std::unique_ptr<ThreadPool>> thread_pool_map;`